src/EncryptStreamWrapper.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace Drupal\file_encrypt;
+
+use Drupal\Core\Site\Settings;
+use Drupal\Core\StreamWrapper\LocalStream;
+use Drupal\Core\StreamWrapper\StreamWrapperInterface;
+use Drupal\Core\Url;
+
+class EncryptStreamWrapper extends LocalStream {
+
+  const SCHEME = 'encrypt';
+
+  protected $fileInfo;
+  protected $mode;
+  protected $encryptionProfile;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getType() {
+    return StreamWrapperInterface::LOCAL_NORMAL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getName() {
+    return t('Encrypted files');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDescription() {
+    return t('Encrypted local files served by Drupal.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDirectoryPath() {
+    return static::basePath();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExternalUrl() {
+    $path = str_replace('\\', '/', $this->getTarget());
+    return Url::fromRoute('system.encrypted_file_download', ['filepath' => $path], ['absolute' => TRUE])
+      ->toString(TRUE)->getGeneratedUrl();
+  }
+
+  /**
+   * Returns the base path for encrypted://.
+   *
+   * Note that this static method is used by \Drupal\system\Form\FileSystemForm
+   * so you should alter that form or substitute a different form if you change
+   * the class providing the stream_wrapper.private service.
+   *
+   * @return string
+   *   The base path for private://.
+   */
+  public static function basePath() {
+    return Settings::get('encrypted_file_path', '');
```

Comment:
For me this is a similar setting to 
```
$settings['file_private_path'] = '';
$settings['file_public_path'] = 'sites/default/files';
```
so something which is environment specific, for which ```settings.php``` is a good location.

Do you think having this in actual configuration would be better?